biscuit.go

```diff
@@ -159,6 +160,39 @@ func (b *Biscuit) Serialize() ([]byte, error) {
 	return proto.Marshal(b.container)
 }
 
+// Sha256Sum returns a hash of `count` biscuit blocks + the authority block
+// along with their respective keys.
+func (b *Biscuit) Sha256Sum(count int) ([]byte, error) {
+	if count < 0 {
+		return nil, fmt.Errorf("biscuit: invalid count,  %d < 0 ", count)
+	}
+	if g, w := count, len(b.container.Blocks); g > w {
+		return nil, fmt.Errorf("biscuit: invalid count,  %d > %d", g, w)
+	}
+
+	partialContainer := &pb.Biscuit{
+		Authority: b.container.Authority,
+		Blocks:    b.container.Blocks[:count],
+		Keys:      b.container.Keys[:count+1], // +1 for the root key
+	}
+
+	s, err := proto.Marshal(partialContainer)
```

Comment:
We need to feed the raw biscuit block bytes read from the outer proto directly into the hash function because protobuf encoding isn't guaranteed to be deterministic.